### PR TITLE
[ButtonBar] Remove UIAppearance call in themer

### LINKED ISSUE #2789: [ButtonBar] MDCButtonBarColorThemer calls UIAppearance internally
### Overview
`MDCButtonBarColorThemer` calls UIAppearance to affect button bars within NavigationBars. This can have unintended consequences for callers who expect it to only theme a specific instance of a button bar (e.g., a toolbar).

### Version number
43.1.0

### PR DESCRIPTION
**Pull request by romoore**
MDCButtonBarColorThemer was explicitly calling UIAppearance for MDCButtonBar when contained in an MDCNavigationBar to clear the backgroundColor property. This is "magic" behavior and should not be included in the themer. Instead, the navigation bar will force the buttonBar background colors to `nil` (clear).  Since having ButtonBars is an implementation detail, clients should not expect them to have a different background color than the navigation bar.

Closes #2789

## Potential client changes
Any clients that are currently using UIAppearance to manipulate the ButtonBar `backgroundColor` property when contained within a UINavigationBar will stop being able to do so. Although this was the solution previously used by the MDCButtonBarColorThemer, it is actually an implementation detail of MDCNavgationBar (not supporting colors button bars) and should not be relied upon by other components.

### CODE REVIEW

**Comment by romoore** [comment]
Marking as "Breaking API change" because this changes internal behavior, even if the API remains the same.

**Review by ianegordon — DISMISSED** [review]

**Comment by ianegordon** [comment]
Please land when Travis gives a +1.

**Comment by romoore** [comment]
Want to get a +1 from @yarneo  (component owner).

**Review by ajsecord — DISMISSED** [review]
Can you update the description to say what breakage is expected because of this change, and what clients should do about it? I.e. "Clients should expect to see X instead of Y, and can add Z to fix the issue."

**Comment by romoore** [comment]
@ajsecord Updated, please take a look.

**Comment by romoore** [comment]
Oops, wrong button.

**Review by ajsecord — DISMISSED** [review]

**Comment by mohammadcazig** [comment]
Suggestion:
We could also make MDCNavigationBar responsible for coloring the background of buttons inside it.

**Comment by romoore** [comment]
@mohammadcazig After thinking through this a bit more, I don't think it makes sense for the button bars to have a separate background color. If in the future we wanted the MDCNavigationBar to host the buttons as subviews directly, it would make it hard to maintain the API.  I think it's best for the internal view hierarchy of the MDCNavigationBar to remain as an implementation detail.

File: catalog/MDCCatalog/MDCCatalogThemer.m [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  if ([NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:iOS9Version]) {
+    MDCButtonBar *appearanceProxy =
+        [MDCButtonBar appearanceWhenContainedInInstancesOfClasses:@[MDCNavigationBar.class]];
+    appearanceProxy.backgroundColor = nil;
+  } else {
+#endif
+    [MDCButtonBar appearanceWhenContainedIn:MDCNavigationBar.class, nil].backgroundColor = nil;
+#if defined(__IPHONE_9_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
+  }
+#endif
```
Comment (mohammadcazig) [inline catalog/MDCCatalog/MDCCatalogThemer.m]: Do you still need this code? since you are setting the background color to nil?

Comment (romoore) [reply catalog/MDCCatalog/MDCCatalogThemer.m]: Nope, good catch. Thanks!

**Comment by yarneo** [comment]
Is there a reason why you created the MDCCatalogThemer with ObjC? If the entire catalog is written in Swift currently it might make better sense to leave it like that than adding a bridging header to accommodate the new code.

**Comment by romoore** [comment]
I didn't have to add a bridging header (it was already present), but I did write in ObjC originally because I was using the `appearanceWhenContainedIn` variadiac version for iOS 8. Now that we're no longer using UIAppearance containment, I can revert the changes back to the AppDelegate.

**Review by yarneo — APPROVED** [review]

**Pull request merged**